pkg/trace/remoteconfighandler/remote_config_handler.go

```diff
@@ -115,6 +141,14 @@ func (h *RemoteConfigHandler) onAgentConfigUpdate(updates map[string]state.RawCo
 		}
 	}
 
+	if len(mergedConfig.CustomTags) > 0 && !(mapsAreEqual(mergedConfig.CustomTags, h.configState.LatestCustomTags)) {
```

Comment:
I don't think our merge logic is aware of your new config item - it's probably more work than worth it to make it aware.

And we're going to take a shortcut and hard-code the configID of configs targeting CustomTags.

1. Check to see if `updates` has the config ID we've selected for hard coding.
2. If so, json.Unmarshal the `RawConfig.Config` part to extract your config data.
3. Then compare that to what is stored in `configState` just like you are doing.

Not much different, we're just not relying on the whole `mergedConfig` bit.

On a similar note, if `updates` does NOT have the config ID we selected you'll want to set the custom tags back to whatever the standard config is. A config ID not being present in the `updates` map means it was deleted. (we don't give these callback functions only what changed, we give it the whole current state, `updates` is probably not the best variable name on our part, oops....)